### DISCUSSION TITLE
compliance check error on my PR

### DISCUSSION THREAD

**Discussion by mkaricheri** [zephyrproject-rtos/zephyr#65825] [Q&A]
Hello,

I see that my PR here https://github.com/zephyrproject-rtos/zephyr/pull/65250 results in a compliance check error below. I had run checkpatch.pl and clang-format on the commits and the c file. Still complaining about coding-style issue without any hint. Also complains about Missing newline. If I add a new line, it complains about extra newline at line 71 of the README.rst. Can someone help me understand the issue here?

Run if [[ ! -s "compliance.xml" ]]; then
Error: See https://docs.zephyrproject.org/latest/contribute/guidelines.html#coding-style for more details.

 Missing newline at end of 'samples/drivers/soc_flash/README.rst'. Check your text editor settings.
Error: Process completed with exit code 1.

**Comment by mkaricheri** [comment]
> Hello,
> 
> I see that my PR here #65250 results in a compliance check error below. I had run checkpatch.pl and clang-format on the commits and the c file. Still complaining about coding-style issue without any hint. Also complains about Missing newline. If I add a new line, it complains about extra newline at line 71 of the README.rst. Can someone help me understand the issue here?
> 
> Run if [[ ! -s "compliance.xml" ]]; then Error: See https://docs.zephyrproject.org/latest/contribute/guidelines.html#coding-style for more details.
> 
> Missing newline at end of 'samples/drivers/soc_flash/README.rst'. Check your text editor settings. Error: Process completed with exit code 1.

Is newline required at the end of README.rst file? Where are the rules written that I can refer for fixing the compliance issue? Thanks

**Reply by erwango** [reply]
> Is newline required at the end of README.rst file? Where are the rules written that I can refer for fixing the compliance issue? Thanks

Actually, you don't need to create a new commit to fix it, so no need to refer to the rule.
You just need to edit the fault commit.

**Reply by mkaricheri** [reply]
@erwango I don't get it.  I actually fixed the failing on commit 948c035e4adda89ac35dcc4083f61aedfdd95d62 which adds README.rstby adding a new line.  This is the diff after that push where I removed the new line again https://github.com/zephyrproject-rtos/zephyr/compare/e556ea01d8a5508e78840831ffb280008bdab093..023b15a3c6b106e4544b44c9230e4f13a5eb8862 which showed it had an extra line at the end. But it gave an error that time. May be I had some additional white spaces in the new line.   But today I repeated the same and compliance check is now passing. So this thread can be closed.

**Comment by mkaricheri** [comment]
> > Is newline required at the end of README.rst file? Where are the rules written that I can refer for fixing the compliance issue? Thanks
> 
> Actually, you don't need to create a new commit to fix it, so no need to refer to the rule. You just need to edit the fault commit. 
I had actually edited and fixed the faulty commit. May be I had some additional white spaces in the new line. But today I repeated the same and compliance check has passed. So this thread can be closed
